### PR TITLE
chore: Adds check for license header

### LINKED ISSUE #3192: chore: add custom rule to check the existence of license header in each file
A rule for checking the existence of a header needs to be added to eslint

### PR DESCRIPTION
**Pull request by konstantinabl**
**Description**:

#### Changes

Added a new package eslint-plugin-header that implements a rule to check the presence of a particular header. 

#### Technical Details

The rule allow for the usage of regex for certain parts of the header. In our case I have added a regex for the line containing dates.
The regex will work for strings with a single date:

```
 * Copyright (C) 2024 Hedera Hashgraph, LLC

```
or a dates defining some period

```
 * Copyright (C) 2022-2024 Hedera Hashgraph, LLC
```

**Related issue(s)**:

Fixes #3192 

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by acuarica — APPROVED** [review]
lgtm

File: .eslintrc.js [hunk 1]
```diff
@@ -64,5 +67,26 @@ module.exports = {
         }],
         "simple-import-sort/imports": "error",
         "simple-import-sort/exports": "off",
+        "header/header": [2, "block", [
+            "-",
+            " *",
+            " * Hedera JSON RPC Relay",
+            " *",
+            {"pattern": "^\\s\\*\\sCopyright \\(C\\) 20(1[8-9]|[2-9]\\d)(?:-20(1[8-9]|[2-9]\\d))? Hedera Hashgraph, LLC$", "template": ` * Copyright (C) ${new Date().getFullYear()} Hedera Hashgraph, LLC`},
```
Comment (Nana-EC) [inline .eslintrc.js:75]: Just making sure, this will manage the year ranges also e.g. 2022 - 2024 for older file.

**Review by Nana-EC — APPROVED** [review]
LG, one question to confirm

File: .eslintrc.js (hunk 1 above)
Comment (konstantinabl) [reply .eslintrc.js:75]: Yes, it manages year ranges and single year

**Pull request merged**